### PR TITLE
Catalog Modal Card Filters Only

### PR DESCRIPTION
**Pull request by gilhsu**
Filter Resource Cards By:
- `!showCollectionOnly` && `!showMarkedOnly`
- `showCollectionOnly` && `!showMarkedOnly`
- `!showCollectionOnly` && `showMarkedOnly`
- `showCollectionOnly` && `showMarkedOnly`

Add disabled state for `MarkedIcon` and `CollectionIcon`.
Add `markedResources` state per collection
Deleted top-level `markedResources` state

https://user-images.githubusercontent.com/45667486/114076082-3e131b80-9874-11eb-8c6a-fc4685c0c998.mov

### CODE REVIEW

File: src/redux/selectors/index.js [hunk 1]
```diff
@@ -384,3 +388,88 @@ export const timelineIntervalsSelector = createSelector(
...
+    (_, ownProps) => ownProps,
+  ],
+  (
+    filteredResourceTypes,
+    selectedResourceType,
+    showCollectionOnly,
+    showMarkedOnly,
+    ownProps,
+  ) => {
+    const { fromContentPanel, fromCatalogScreen } = ownProps;
```
Comment (foxtree) [inline src/redux/selectors/index.js]: is it valid to have a `fromContentPanel` and  `fromCatalogScreen` value at the same time?   (or: if one is false, the other must be true?)

if not, I wonder if there should be a single variable.

Comment (gilhsu) [reply src/redux/selectors/index.js]: makes sense. I can refactor.

File: src/components/SubTypeAccordion/SubTypeAccordionsContainer.js [hunk 2]
```diff
@@ -36,16 +33,11 @@ const SubTypeAccordionsContainer = ({ subTypeData, fromContentPanel }) => (
...
-
-SubTypeAccordionsContainer.defaultProps = {
-  fromContentPanel: false,
+  accordionsContainerData: shape({}).isRequired,
 };
 
-const mapStateToProps = (state) => ({
-  resourceTypeFilters: state.resourceTypeFilters,
+const mapStateToProps = (state, ownProps) => ({
+  accordionsContainerData: accordionsContainerDataSelector(state, ownProps),
```
Comment (foxtree) [inline src/components/SubTypeAccordion/SubTypeAccordionsContainer.js:40]: it seems `fromContentPanel` and `fromCatalogScreen` escape propType validation, when used via `mapStateToProps`

I wonder if there's an eslint rule that can detect this via the caller/parent component that passes the props.

Comment (gilhsu) [reply src/components/SubTypeAccordion/SubTypeAccordionsContainer.js:40]: couldn't find a rule upon quick google search. perhaps this is a refactor later. Or I can add the props validation and add `react/no-unused-prop-types` with a note

**Review by foxtree — APPROVED** [review]

**Pull request merged**